src/keystore.c

```diff
@@ -427,6 +427,23 @@ static bool _get_xprv(const uint32_t* keypath, const size_t keypath_len, struct
     return true;
 }
 
+bool keystore_get_root_fingerprint(uint8_t* fingerprint)
+{
+    if (sizeof(fingerprint) < sizeof(uint32_t)) {
+        Abort("fingerprint buffer needs to be size 4");
+    }
+    static const uint32_t keypath[] = {
+        0 + BIP32_INITIAL_HARDENED_CHILD,
+    };
+
+    struct ext_key derived_xpub __attribute__((__cleanup__(keystore_zero_xkey))) = {0};
+    if (!keystore_get_xpub(keypath, sizeof(keypath) / sizeof(uint32_t), &derived_xpub)) {
+        return false;
+    }
+    memcpy(fingerprint, derived_xpub.parent160, sizeof(uint32_t));
```

Comment:
not sure `sizeof(uint32_t)` is useful, as there is no `uint32_t` type used anywhere. I'd just put `4` here.

Reply:
It's what libwally uses to calculate the fingerprint that is encode with each extended key. I also found more `sizeof(uint32_t)` than `4`, so I thought I was being consistent here.

Reply:
Anyway, changed it to `4` now, since it is more readable.